lib/engine/game/g_1882.rb

```diff
@@ -39,6 +39,8 @@ class G1882 < Base
       # Two lays or one upgrade, second tile costs 20
       TILE_LAYS = [{ lay: true, upgrade: true }, { lay: :not_if_upgraded, upgrade: false, cost: 20 }].freeze
 
+      PREPRINT_COLOR_ON_BORDER = %w[F12].freeze
```

Comment:
Could this auto-calculate the preprinted tiles as part of the base class?

Reply:
Maybe, but i was feeling lazy.

I think that ends up being more expensive, every hex with track on it would need to compare with every rotation of every tiles until it finds a match--in 1882's case the match should bearly since F12 is like a tile 8--18 Los Angeles has some yellow cities including a higher-numbered 57. I guess you'd first sort the tiles by color instead of just number... :thinking:

But then all the hexes with track that _don't_ match a tile _do_ have to check against all the tiles in the manifest.

Maybe I could finally learn how to use stackprof to investigate this :slightly_smiling_face:

Reply:
Would it not be easier to make all 'printed' upgradable tiles have a border that matches the base color?

Reply:
This is the new behavior at 077560ae, here are some screenshots:

![Screenshot from 2020-11-20 22-37-52](https://user-images.githubusercontent.com/1045173/99868592-194a3000-2b81-11eb-9334-cdd1ceac22d6.png)
![Screenshot from 2020-11-20 22-37-37](https://user-images.githubusercontent.com/1045173/99868593-19e2c680-2b81-11eb-9080-6d493f6c2a7d.png)
![Screenshot from 2020-11-20 22-37-29](https://user-images.githubusercontent.com/1045173/99868595-19e2c680-2b81-11eb-87db-9618d5a39c85.png)